cli/dcoscli/node/main.py

```diff
@@ -472,9 +472,21 @@ def _list(json_, extra_field_names):
     """
 
     client = mesos.DCOSClient()
+    masters = mesos.MesosDNSClient().hosts('master.mesos.')
+    master_state = client.get_master_state()
     slaves = client.get_state_summary()['slaves']
+    for master in masters:
+        if master['ip'] == master_state['hostname']:
+            master['type'] = 'master (leader)'
+            for key in ('id', 'pid', 'version'):
+                master[key] = master_state.get(key)
```

Comment:
This does add it to the `json` view too though.

```
$ dcos node --json | jq '.[0:3]'
[
  {
    "host": "master.mesos.",
    "id": "74df965e-a46e-46a9-8d2a-639db0af0e25",
    "ip": "10.10.12.68",
    "pid": "master@10.10.12.68:5050",
    "type": "master (leader)",
    "version": "1.0.1"
  },
  {
    "host": "master.mesos.",
    "ip": "10.10.11.179",
    "type": "master"
  },
  {
    "host": "master.mesos.",
    "ip": "10.10.13.72",
    "type": "master"
  }
]
```

I think I must be missing your point.